### PR TITLE
[tune] add accessible trial_info

### PR DESCRIPTION
**Pull request by richardliaw**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/tune/trainable.py [hunk 1]
```diff
@@ -511,6 +515,18 @@ def logdir(self):
         """
         return os.path.join(self._logdir, "")
 
+    @property
+    def trial_info(self):
+        """TrialInfo object for the corresponding trial of this Trainable.
+
+        This allows you to access fields such as the trial_id and trial name
+        inside the training loop. See the documentation on TrialInfo for more
+        informatino.
```
Comment (ujvl) [inline python/ray/tune/trainable.py]: wait why allow the ID and name to change? shouldn't these be read-only

also typo (information)

Comment (richardliaw) [reply python/ray/tune/trainable.py]: Fixed; changed to be exposed as properties.

File: doc/source/tune-package-ref.rst [hunk 2]
```diff
@@ -18,6 +18,9 @@ ray.tune
 .. autoclass:: ray.tune.function_runner.StatusReporter
     :members: __call__, logdir
 
+
+.. autoclass:: ray.tune.trial.TrialInfo
```
Comment (richardliaw) [inline doc/source/tune-package-ref.rst]: todo

File: doc/source/tune-package-ref.rst [hunk 3]
```diff
@@ -18,6 +18,9 @@ ray.tune
...
+
```
Comment (richardliaw) [inline doc/source/tune-package-ref.rst]:
```suggestion
```

File: python/ray/tune/function_runner.py [hunk 4]
```diff
@@ -133,7 +148,10 @@ def _setup(self, config):
         self._error_queue = queue.Queue(1)
 
         self._status_reporter = StatusReporter(
-            self._results_queue, self._continue_semaphore, self.logdir)
+            self._results_queue,
+            self._continue_semaphore,
+            trial_info=self.trial_info,
```
Comment (richardliaw) [inline python/ray/tune/function_runner.py]:
```suggestion
            trial_info=self._trial_info,
```

**Review by ujvl — APPROVED** [review]

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by richardliaw** [comment]
jenkins test tune

**Pull request merged**